crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -1048,7 +1055,12 @@ impl CumulocityConverter {
 
                 return Ok(mapped_messages);
             }
-            Ok(_) => {}
+            Ok(_) => {
+                // Handle the case where @id is missing but there are no other changes to the payload
+                let mut updated_metadata_message =
+                    self.append_id_if_not_given(register_message, &register_message.topic_id);
+                mapped_messages.append(&mut updated_metadata_message);
```

Comment:
So this is the case a registration message forged by some external software (running on behalf of a child device).

- Is this okay for C8Y to have an external id that is not derived from the main device name (as done here)?
- Is this compatible with auto-registration (say a measurement is received from this child device *before* its registration)?
- If auto-registration is disabled and no @id is provided by the registration message, what would be the external id for the device?

Reply:
This is the case where a registration message is arrived, but it's judged that the message affects no entities...

> Is this okay for C8Y to have an external id that is not derived from the main device name (as done here)?

Should be ok.

> Is this compatible with auto-registration (say a measurement is received from this child device before its registration)?

If some entities are registered during the auto-registration process, the new registration messages will be published. Then, these messages are arrived in this condition arm exactly, and republishes (=overwrite) the retained messages with `@id`.

> If auto-registration is disabled and no @id is provided by the registration message, what would be the external id for the device?

I think the message goes to "pending entities". @albinsuresh can you confirm?

Reply:
> If some entities are registered during the auto-registration process, the new registration messages will be published. Then, these messages are arrived in this condition arm exactly, and republishes (=overwrite) the retained messages with @id.

For auto-registered entities, their generated registration messages that are published should already have the `@id` field in it. So, it shouldn't need another republish.

>  think the message goes to "pending entities". @albinsuresh can you confirm?

That's correct. They are kept cached until the registration message arrives (with or without the `@id`).